scripts/DllManipulatorScript.cs

```diff
@@ -110,10 +130,31 @@ public static void InvokeMainThreadQueue()
         }
 
 #if UNITY_EDITOR
+        private bool _isRecompiling;
+        /// <summary>
+        /// Called when Assemblies are reloaded due to recompilation.
+        /// Called before OnDisable.
+        /// </summary>
+        private void OnBeforeAssemblyReload()
+        {
+            _isRecompiling = true;
+        }
+        
         private void OnDisable()
         {
-            if(!EditorApplication.isPlaying && Options.enableInEditMode)
+            if(_singletonInstance == this && !EditorApplication.isPlaying && Options.enableInEditMode)
+            {
                 EditorApplication.update -= Update;
+                AssemblyReloadEvents.beforeAssemblyReload -= OnBeforeAssemblyReload;
+                
+                // When recompiling OnDestroy is not called by default (the object is not really destroyed)
+                // Manually trigger OnDestroy to clean up if we are disabled because of recompilation
+                if (_isRecompiling)
+                {
+                    _isRecompiling = false;
+                    OnDestroy();
```

Comment:
I don't really like calling a callback method so I'd be after creating separate one

Reply:
Created a Reset()